### ISSUE TITLE
No global error handler

### ISSUE THREAD

**Issue by xule1991** [redux-observable/redux-observable#619]
**Do you want to request a *feature* or report a *bug*?**  
I want to request for a new feature

**What is the current behavior?**
no global error handler

**If the current behavior is a bug, please provide the steps to reproduce and a minimal demo of the problem using JSBin, StackBlitz, or similar.**

**What is the expected behavior?**
when stream encounters an error in catchError, it will dispatch a GLOBAL_ERROR action or expose an interface to let user to add a global error handler

**Which versions of redux-observable, and which browser and OS are affected by this issue? Did this work in previous versions of redux-observable?**

**Comment by zepod** [comment]
Epics are input/output functions of actions. I don't think it is desirable to have them dispatch foreign actions by themselves.

It Is true however, it Is awkward having to add `catchError(globalHandler)` to every single epic. 

Possible solution would be to append multiple epics at the point of `combineEpics` with desposable stream, that would catch errors that are not catched within the epic.

Not sure if such a helper function should be part of this library or should be a distinct one.

I'd be nice to have more opinions on this, so either pull request or distinct library could be made.

**Comment by ThomasLabstep** [comment]
```
export const onSomethingActionEpic = (
  action$: Observable<Action>,
): Observable<Action> =>
  action$.pipe(
    filter((action: Action) => action.type === WHATEVER),
    tap((action: Action) => {
      // Do something
    }),
    ignoreElements(),
    catchError((err: any, source$: Observable<any>) => concat(
      of({ type: 'EPIC_FAIL', payload: err }),
      source$,
    )),
  );
```

This catchError per epic is working.
And the concat allows redux-observable to keep working.
But it's impossible to unit test it.

```
export const epicError = (epic, errorType: string) => {
  testScheduler.run(({ hot, cold, expectObservable, expectSubscriptions }) => {
    const action$ = hot('#--|');

    const output$ = epic(action$);

    expectObservable(output$).toBe('a--|', {a: {
      payload: 'error',
      type: errorType,
    }});
  });
};
```

The following will raise Maximum call stack size exceeded.

I thought maybe we could limit to take(1) but haven't figure out how.

**Comment by punksta** [comment]
https://github.com/redux-observable/redux-observable/issues/591 might be related

**Comment by paubor** [comment]
Sorry I linked the wrong docs: I mean "Adding a global error handler" in https://redux-observable.js.org/docs/basics/SettingUpTheMiddleware.html

**Comment by evertbouw** [comment]
@paubor Yes that article is still relevant. I don't think the library needs to provide anything that you can easily set up to your liking with rxjs. I don't really want to close issues because I welcome discussions and adding a comment is a lower barrier than opening new issues

**Comment by MikeRyanDev** [comment]
Closing this as won't fix given how old the issue is. If there's something still missing here, please open a new issue! 😃
